third_party/move/move-compiler-v2/src/bytecode_generator.rs

```diff
@@ -856,6 +856,12 @@ impl<'env> Generator<'env> {
                 );
                 let target_ty = self.temp_type(targets[0]).clone();
                 if let Type::Struct(wrapper_mid, wrapper_sid, wrapper_inst) = target_ty {
+                    if wrapper_mid != *mid {
+                        self.error(
+                            id,
+                            "cannot implicitly pack a wrapper struct defined in a different module",
```

Comment:
Because this is a new feature, we should ideally provide more information. Perhaps something that would look like: "cannot implicitly pack wrapper struct `m::Lazy` defined in a different module `0xc0ffee::m`"

Similarly for the unpack case (even though it was fixed in a different PR).

Reply:
updated the error message